### ISSUE TITLE
Wrong architecture for 22.10.0 macosx wheel

### ISSUE THREAD

**Issue by lecardozo** [psf/black#3312]
**Describe the bug**

I've just recently installed the latest version of Black (22.10.0) on a Python 3.10.5 environment, Mac OSX Monterey 12.5 and got the following error when trying to run
```shell
➜  black
Traceback (most recent call last):
  File "/Users/me/.pyenv/versions/testing-black/bin/black", line 5, in <module>
    from black import patched_main
ImportError: dlopen(/Users/me/.pyenv/versions/3.10.5/envs/testing-black/lib/python3.10/site-packages/black/__init__.cpython-310-darwin.so, 0x0002): tried: '/Users/me/.pyenv/versions/3.10.5/envs/testing-black/lib/python3.10/site-packages/black/__init__.cpython-310-darwin.so' (mach-o file, but is an incompatible architecture (have (arm64), need (x86_64h)))
```
After investigating the compiled file, it does seem that it was built for the wrong architecture (arm64 instead of x86_64)

```
➜ file /Users/me/.pyenv/versions/3.10.5/envs/testing-black/lib/python3.10/site-packages/black/__init__.cpython-310-darwin.so
/Users/me/.pyenv/versions/3.10.5/envs/testing-black/lib/python3.10/site-packages/black/__init__.cpython-310-darwin.so: Mach-O 64-bit bundle arm64
```
**To Reproduce** 

For example, take this code:

```python
➜ pyenv install python 3.10.5
➜ pyenv virtualenv 3.10.5 testing-black
➜ pyenv activate testing-black
➜ pip install --upgrade pip && pip install black==22.10.0
➜ black
```
The resulting error should be similar to the one reported above.

To make sure this is no pip mistake (maybe downloading the wrong package?) I've directly downloaded the [.whl from Pypi](https://files.pythonhosted.org/packages/6f/9b/e739926908b08a4e105d8cfdaaa0345543cbe770b8909607740e1e79fc63/black-22.10.0-cp310-cp310-macosx_11_0_x86_64.whl) and could reproduce the issue.

**Expected behavior**

The pre-built wheel should be compiled to the correct architecture, I guess.

**Environment**

- Black's version: 22.10.0
- OS and Python version: MacOS Monterey 12.5 (x86_64) / Python 3.10.5

**Additional context**

**Comment by ichard26** [comment]
Ugh, this is a known issue. I messed up and this release only shipped with compiled wheels for macOS x86-64 due to a build configuration issue I presume. Looking at your bug report carefully, the wheels actually built fine themselves, but their platform tags are incorrect. 

@JelleZijlstra has deleted the macOS wheels to avoid breakage while we figure this out.

**Comment by JelleZijlstra** [comment]
Yes, we ended up uploading amd64 wheels labeled as x86_64. I confirmed that after deleting the wheels from PyPI, installing Black 22.10.0 on MacOS x86_64 works fine; it just uses the pre Python wheel.

**Comment by ichard26** [comment]
I fixed the wheel metadata for all x86_64 and arm64 wheels manually and have uploaded them to PyPI. I have not fixed the universal2 wheels, if people want/need them, please holler.

**Comment by JelleZijlstra** [comment]
For the 22.12.0 release I skipped the amd64 and universal2 MacOS wheel builds to avoid this issue, but the x86_64 job also failed because apparently it was trying to test the x86_64 wheel on an amd64 machine (https://github.com/psf/black/actions/runs/3658832698/jobs/6184104544). For now we'll have no mypyc-compiled macos wheels.

**Comment by cooperlees** [comment]
arm64? amd64 is x86_64 …

**Comment by JelleZijlstra** [comment]
Uh yeah, I always confuse them. a<something>64 :)

**Comment by ichard26** [comment]
[Fixed since 23.1.0](https://pypi.org/project/black/23.1.0/#files).
